### PR TITLE
[cli] [topo] Migrate topo flags to pflags

### LINKED ISSUE #10857: Switch flag definitions to be on pflag instead of flag in `package go/vt/topo`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/topo/*.go
go/vt/topo/locks.go:    RemoteOperationTimeout = flag.Duration("remote_operation_timeout", 30*time.Second, "time to wait for a remote operation")
go/vt/topo/server.go:   topoImplementation = flag.String("topo_implementation", "", "the topology implementation to use")
go/vt/topo/server.go:   topoGlobalServerAddress = flag.String("topo_global_server_address", "", "the address of the global topology server")
go/vt/topo/server.go:   topoGlobalRoot = flag.String("topo_global_root", "", "the path of the global topology data in the global topology server")
```

### PR DESCRIPTION
**Pull request by rsajwani**
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description
Updates topo related flag references specified in https://github.com/vitessio/vitess/issues/10857

endpoints using these flags 
```
$ go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ "vitess.io/vitess/go/vt/topo" {print $1}'
vitess.io/vitess/go/cmd/mysqlctl
vitess.io/vitess/go/cmd/mysqlctld
vitess.io/vitess/go/cmd/rulesctl
vitess.io/vitess/go/cmd/rulesctl/cmd
vitess.io/vitess/go/cmd/rulesctl/common
vitess.io/vitess/go/cmd/topo2topo
vitess.io/vitess/go/cmd/vtadmin
vitess.io/vitess/go/cmd/vtbackup
vitess.io/vitess/go/cmd/vtbench
vitess.io/vitess/go/cmd/vtcombo
vitess.io/vitess/go/cmd/vtctl
vitess.io/vitess/go/cmd/vtctld
vitess.io/vitess/go/cmd/vtctldclient
vitess.io/vitess/go/cmd/vtctldclient/cli
vitess.io/vitess/go/cmd/vtctldclient/command
vitess.io/vitess/go/cmd/vtctldclient/docgen
vitess.io/vitess/go/cmd/vtexplain
vitess.io/vitess/go/cmd/vtgate
vitess.io/vitess/go/cmd/vtgateclienttest
vitess.io/vitess/go/cmd/vtgr
vitess.io/vitess/go/cmd/vtorc
vitess.io/vitess/go/cmd/vttablet
vitess.io/vitess/go/cmd/vttestserver
vitess.io/vitess/go/cmd/zk
```
## Related Issue(s)
closes #10857

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/topo/server.go [hunk 1]
```diff
@@ -176,6 +174,19 @@ var (
 	}
 )
 
+func init() {
+	for _, cmd := range []string{"vttablet", "vtctl", "vtctld", "vtcombo", "vtexplain", "vtgate",
+		"vtgr", "vtorc", "vtbackup"} {
```
Comment (ajm188) [inline go/vt/topo/server.go:179]: are the binaries for each file all the same? if so we should put them in one variable so we don't inevitably forget to update just one instance if we ever change this

Comment (rsajwani) [reply go/vt/topo/server.go:179]: are you suggesting to have a constant [] across lock and server file?

File: go/flags/endtoend/vttestserver.txt [hunk 2]
```diff
@@ -128,9 +127,6 @@ Usage of vttestserver:
       --topo_etcd_tls_ca string                                          path to the ca to use to validate the server cert when connecting to the etcd topo server
       --topo_etcd_tls_cert string                                        path to the client cert to use to connect to the etcd topo server, requires topo_etcd_tls_key, enables TLS
       --topo_etcd_tls_key string                                         path to the client key to use to connect to the etcd topo server, enables TLS
-      --topo_global_root string                                          the path of the global topology data in the global topology server
-      --topo_global_server_address string                                the address of the global topology server
-      --topo_implementation string                                       the topology implementation to use
```
Comment (deepthi) [inline go/flags/endtoend/vttestserver.txt:133]: I'd expect vttestserver to have all the same flags as vttablet + vtgate + vtctld.
So it doesn't look right to remove these here.

Comment (deepthi) [reply go/flags/endtoend/vttestserver.txt:133]: or maybe they are hard-coded?

**Review by deepthi — APPROVED** [review]

File: go/flags/endtoend/vttestserver.txt (hunk 2 above)
Comment (GuptaManan100) [reply go/flags/endtoend/vttestserver.txt:122]: Vttestserver spwans the vtcombo process, but for whatever reason it uses flags `external_topo_implementation`, etc. It passes these values along to vtcombo as `topo_implementation` etc

Comment (GuptaManan100) [reply go/flags/endtoend/vttestserver.txt:122]: Weird I know, but it doesn't use these `topo_implementation` flags itself.

**Pull request merged**